src/EnergyPlus/WeatherManager.cc

```diff
@@ -2028,24 +2010,37 @@ namespace WeatherManager {
 
                     if (state.dataGlobal->DayOfSim < state.dataWeatherManager->curSimDayForEndOfRunPeriod) {
                         ResetWeekDaysByMonth(state,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).MonWeekDay,
+                                             thisEnvrn.MonWeekDay,
                                              state.dataWeatherManager->LeapYearAdd,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).StartMonth,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).StartDay,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).EndMonth,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).EndDay,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).RollDayTypeOnRepeat,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).RollDayTypeOnRepeat ||
-                                                 state.dataEnvrn->CurrentYearIsLeapYear);
+                                             thisEnvrn.StartMonth,
+                                             thisEnvrn.StartDay,
+                                             thisEnvrn.EndMonth,
+                                             thisEnvrn.EndDay,
+                                             thisEnvrn.RollDayTypeOnRepeat,
+                                             thisEnvrn.RollDayTypeOnRepeat || state.dataEnvrn->CurrentYearIsLeapYear);
                         if (state.dataWeatherManager->DaylightSavingIsActive) {
-                            SetDSTDateRanges(state,
-                                             state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).MonWeekDay,
-                                             state.dataWeatherManager->DSTIndex);
+                            SetDSTDateRanges(state, thisEnvrn.MonWeekDay, state.dataWeatherManager->DSTIndex);
                         }
-                        SetSpecialDayDates(state, state.dataWeatherManager->Environment(state.dataWeatherManager->Envrn).MonWeekDay);
+                        SetSpecialDayDates(state, thisEnvrn.MonWeekDay);
                     }
                 }
             }
+
+            // at the beginning of each day find the min/max weather used for DOAS sizing
+            if (thisEnvrn.KindOfEnvrn == DataGlobalConstants::KindOfSim::RunPeriodDesign ||
+                thisEnvrn.KindOfEnvrn == DataGlobalConstants::KindOfSim::DesignDay) {
+                for (size_t i = 0; i < state.dataWeatherManager->TodayOutDryBulbTemp.size(); ++i) {
```

Comment:
Since the result of this (`thisEnvrn.maxCoolingOATSizing` etc) are only used by the AirloopHVAC DOAS, seems we should set a flag to skip this entirely is there isn't one.

Reply:
Air loops are read in well after this point. The only way I can think of to set a flag is to check for the number of DOAS objects prior to this point. Maybe we need a new function that can scan the json inputs for things like that at the beginning of the simulation. I have wanted to do that for some time.

Reply:
That's easy enough with 
`bool anyDOAS = (state.dataInputProcessing->inputProcessor->getNumObjectsFound(state, "AirLoopHVAC:DedicatedOutdoorAirSystem") > 0);`
Just a question of a logical spot to put it.

Reply:
@Myoldmopar the comments/suggestion above is where state.dataInputProcessing->inputProcessor->isInputObjectUsed was born.